### PR TITLE
[ioctl] Build hdwallet import command line into new ioctl

### LINKED ISSUE #3286: [ioctl] build hdwallet import command line into new ioctl
### What would you like to be added
 - Step 1: create `hdwalletimport.go` and `hdwalletimport_test.go` files in `./ioctl/newcmd/hdwallet`. 
 - Step 2: refactor  command  https://github.com/iotexproject/iotex-core/blob/master/ioctl/cmd/hdwallet/hdwalletimport.go in `hdwalletimport.go`
 - Step 3: add unit test in xx_test.go

#### Notes:
1. Use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to construct the Cobra command.
2. [Output package](https://github.com/iotexproject/iotex-core/blob/master/ioctl/output) is deprecated. `e.g.` `output.PrintError` could be replaced with `errors.Wrap`.
3. Global variables (config.ReadConfig  config.DefaultConfigFile etc.) have been wrapped into client interface. Please use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to access them if needed
4. The logic of new code should be similar to [the old one](https://github.com/iotexproject/iotex-core/blob/master/ioctl/cmd/hdwallet/hdwalletimport.go).
5. In each unit test, the results should be checked whether they are as the expected. 

#### Example:
1. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountinfo.go
2. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountdelete.go
3. https://github.com/iotexproject/iotex-core/blob/master/ioctl/newcmd/account/accountcreateadd.go

### Why is this needed
Unit tests couldn't be added for the old ioctl command (./ioctl/cmd)

### PR DESCRIPTION
**Pull request by LuckyPigeon**
# Description

Refactor `hdwalletimport` command in new `ioctl`, with the following note.

* Use [client interface](https://github.com/iotexproject/iotex-core/blob/master/ioctl/client.go) to construct the Cobra command.
* Output package is deprecated, replace it with errors package.
* Replace fmt.Println with cmd.Println
* Refactor unit test to cover the modification.

Fixes #3286 

## Type of change
Please delete options that are not relevant.
- [x] Code refactor or improvement

# How Has This Been Tested?
Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration
- [x] make test
- [x] TestNewHdwalletImportCmd

**Test Configuration**:
- Firmware version: Ubuntu 21.04
- Hardware:
- Toolchain:
- SDK:

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: ioctl/newcmd/hdwallet/hdwalletimport.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
+// Copyright (c) 2022 IoTeX Foundation
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package hdwallet
+
+import (
+	"errors"
```
Comment (huof6829) [inline ioctl/newcmd/hdwallet/hdwalletimport.go:10]: use "github.com/pkg/errors"

File: ioctl/newcmd/hdwallet/hdwalletimport.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+			}
+			if password != passwordAgain {
+				return ErrPasswdNotMatch
+			}
+
+			enctxt := append([]byte(mnemonic), util.HashSHA256([]byte(mnemonic))...)
+			enckey := util.HashSHA256([]byte(password))
+			out, err := util.Encrypt(enctxt, enckey)
+			if err != nil {
+				return output.NewError(output.ValidationError, "failed to encrypting mnemonic", nil)
```
Comment (huof6829) [inline ioctl/newcmd/hdwallet/hdwalletimport.go:82]: don't use output packages. use errors.Wrap

File: ioctl/newcmd/hdwallet/hdwalletimport.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+			}
+
+			if err := os.WriteFile(_hdWalletConfigFile, out, 0600); err != nil {
+				return output.NewError(output.WriteFileError,
```
Comment (huof6829) [inline ioctl/newcmd/hdwallet/hdwalletimport.go:86]: the same

File: ioctl/newcmd/hdwallet/hdwalletimport.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+					fmt.Sprintf("failed to write to config file %s", _hdWalletConfigFile), err)
+			}
+
+			cmd.Println(fmt.Sprintf("Mnemonic phrase: %s\n"+
```
Comment (huof6829) [inline ioctl/newcmd/hdwallet/hdwalletimport.go:90]: use cmd.Printf

File: ioctl/newcmd/hdwallet/hdwallet.go [hunk 5]
```diff
@@ -1,11 +1,24 @@
...
+// Errors
+var (
+	ErrPasswdNotMatch = errors.New("password doesn't match")
+)
+
 // DefaultRootDerivationPath for iotex
 // https://github.com/satoshilabs/slips/blob/master/slip-0044.md
 const DefaultRootDerivationPath = "m/44'/304'"
+
+var _hdWalletConfigFile = config.ReadConfig.Wallet + "/hdwallet"
```
Comment (huof6829) [inline ioctl/newcmd/hdwallet/hdwallet.go:24]: donot use config.ReadConfig.  use client.Config()

File: ioctl/newcmd/hdwallet/hdwalletimport.go [hunk 6]
```diff
@@ -0,0 +1,96 @@
...
+		RunE: func(cmd *cobra.Command, args []string) error {
+			cmd.SilenceUsage = true
+			if fileutil.FileExists(_hdWalletConfigFile) {
+				cmd.Println("Please run 'ioctl hdwallet delete' before import")
+				return nil
+			}
+
+			cmd.Println("Enter 12 mnemonic words you saved, separated by space")
+
+			line, err := client.ReadSecret()
```
Comment (huof6829) [inline ioctl/newcmd/hdwallet/hdwalletimport.go:55]: client.ReadSecret  cannot see the input character.  this is not right.  see https://github.com/iotexproject/iotex-core/blob/773fb82140af8c7d4bf81e38b323e6eec3e75da7/ioctl/cmd/hdwallet/hdwalletimport.go#L57-L61
move these codes into client interface. and call function here

File: ioctl/newcmd/hdwallet/hdwalletimport.go [hunk 7]
```diff
@@ -0,0 +1,96 @@
...
+			}
+
+			enctxt := append([]byte(mnemonic), util.HashSHA256([]byte(mnemonic))...)
+			enckey := util.HashSHA256([]byte(password))
+			out, err := util.Encrypt(enctxt, enckey)
+			if err != nil {
+				return output.NewError(output.ValidationError, "failed to encrypting mnemonic", nil)
+			}
+
+			if err := os.WriteFile(_hdWalletConfigFile, out, 0600); err != nil {
```
Comment (huof6829) [inline ioctl/newcmd/hdwallet/hdwalletimport.go:85]: move line 85~88 into client interface. and call function here

File: ioctl/newcmd/hdwallet/hdwalletimport_test.go [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+	mnemonic := "lake stove quarter shove dry matrix hire split wide attract argue core"
+	password := "123"
+
+	client.EXPECT().SelectTranslation(gomock.Any()).Return("mockTranslationString", config.English).Times(4)
+
+	t.Run("import hdwallet", func(t *testing.T) {
+		client.EXPECT().ReadSecret().Return(mnemonic, nil)
+		client.EXPECT().ReadSecret().Return(password, nil)
+		client.EXPECT().ReadSecret().Return(password, nil)
+		client.EXPECT().WriteConfig()
```
Comment (huof6829) [inline ioctl/newcmd/hdwallet/hdwalletimport_test.go:39]: delete if not use

File: ioctl/newcmd/hdwallet/hdwalletimport_test.go [hunk 9]
```diff
@@ -0,0 +1,57 @@
...
+
+		cmd := NewHdwalletImportCmd(client)
+		_, err := util.ExecuteCmd(cmd)
+		require.NoError(err)
+	})
+
+	t.Run("hdwalletConfigFile exist", func(t *testing.T) {
+		expectedValue := "Please run 'ioctl hdwallet delete' before import"
+		err := os.WriteFile(_hdWalletConfigFile, []byte("123"), 0600)
```
Comment (huof6829) [inline ioctl/newcmd/hdwallet/hdwalletimport_test.go:48]: use tmp dir. see example
https://github.com/iotexproject/iotex-core/blob/773fb82140af8c7d4bf81e38b323e6eec3e75da7/ioctl/newcmd/account/accountdelete_test.go#L33-L35

File: ioctl/newcmd/hdwallet/hdwalletimport_test.go [hunk 10]
```diff
@@ -0,0 +1,57 @@
...
+	client.EXPECT().SelectTranslation(gomock.Any()).Return("mockTranslationString", config.English).Times(4)
+
+	t.Run("import hdwallet", func(t *testing.T) {
+		client.EXPECT().ReadSecret().Return(mnemonic, nil)
+		client.EXPECT().ReadSecret().Return(password, nil)
+		client.EXPECT().ReadSecret().Return(password, nil)
+		client.EXPECT().WriteConfig()
+
+		cmd := NewHdwalletImportCmd(client)
+		_, err := util.ExecuteCmd(cmd)
```
Comment (huof6829) [inline ioctl/newcmd/hdwallet/hdwalletimport_test.go:42]: can check result when return no error ?

File: ioctl/newcmd/hdwallet/hdwalletimport.go [hunk 11]
```diff
@@ -0,0 +1,92 @@
...
+			}
+			if password != passwordAgain {
+				return ErrPasswdNotMatch
+			}
+
+			enctxt := append([]byte(mnemonic), util.HashSHA256([]byte(mnemonic))...)
+			enckey := util.HashSHA256([]byte(password))
+			out, err := util.Encrypt(enctxt, enckey)
+			if err != nil {
+				return errors.Wrap(err, "failed to encrypting mnemonic")
```
Comment (huof6829) [inline ioctl/newcmd/hdwallet/hdwalletimport.go:80]: these can be put into client.WriteFile

File: ioctl/newcmd/hdwallet/hdwalletimport.go [hunk 12]
```diff
@@ -0,0 +1,92 @@
...
+			}
+
+			if err := client.WriteFile(_hdWalletConfigFile, out); err != nil {
```
Comment (huof6829) [inline ioctl/newcmd/hdwallet/hdwalletimport.go:83]: `WriteHdWalletConfigFile(mnemonic, password)` is better

File: ioctl/newcmd/hdwallet/hdwallet.go [hunk 13]
```diff
@@ -1,11 +1,22 @@
...
+// Errors
+var (
+	ErrPasswdNotMatch = errors.New("password doesn't match")
+)
+
 // DefaultRootDerivationPath for iotex
 // https://github.com/satoshilabs/slips/blob/master/slip-0044.md
 const DefaultRootDerivationPath = "m/44'/304'"
+
+var _hdWalletConfigFile string
```
Comment (huof6829) [inline ioctl/newcmd/hdwallet/hdwallet.go:22]: try not to use `_hdWalletConfigFile` as a global variable.  put it into client

**Review by CoderZhi — APPROVED** [review]

**Comment by huof6829** [comment]
Don't modify the struct Config in  ioctl/config.go because unknown errors may occur in currently using ioctl/cmd. I'v put it into client.go and add correspond functions for ioctl/newcmd. Pls check it. @LuckyPigeon

**Review by huof6829 — APPROVED** [review]

**Pull request merged**